Python/Product/PythonTools/PythonTools/Extensions.cs

```diff
@@ -911,5 +918,32 @@ public static class Extensions {
             }
             return prettyPrinted.ToString().Trim();
         }
+
+        public static void RunSynchronouslyOnUIThread(this System.Threading.Tasks.Task task, IServiceProvider serviceProvider, [CallerMemberNameAttribute] string callerMemberName = "") {
+            if (!Dispatcher.CurrentDispatcher.CheckAccess()) {
+                string message = string.Format("{0} must be called on a UI thread.", callerMemberName);
+                Debug.Fail(message);
+                return;
```

Comment:
Yes. Essentially, the difference is that `WaitAndUnwrapExceptions` blocks the thread on which it's called, so if the task that it is waiting on requires posting a continuation to that thread at any point, everything deadlocks. Whereas this VS task factory stuff does something (I'm not sure exactly what it is, but I believe it's the absolute minimum that's needed to process incoming continuations) that lets this all work without deadlocks.